go/vt/sqlparser/ast_funcs.go

```diff
@@ -490,7 +490,7 @@ func (node *Literal) encodeHexValToMySQLQueryFormat() ([]byte, error) {
 	}
 
 	// Let's make this idempotent in case it's called more than once
-	if nb[0] == 'x' && nb[1] == '0' && nb[len(nb)-1] == '\'' {
+	if len(nb) > 2 && nb[0] == 'x' && nb[1] == '\'' && nb[len(nb)-1] == '\'' {
```

Comment:
nit: can use regex pattern matching.

Reply:
I thought that might be less efficient but it's certainly simpler so I'll switch to that.